eclair-core/src/main/scala/fr/acinq/eclair/transactions/CommitmentSpec.scala

```diff
@@ -33,8 +33,21 @@ sealed trait CommitmentOutput
 object CommitmentOutput {
   case object ToLocal extends CommitmentOutput
   case object ToRemote extends CommitmentOutput
+  case class InHtlc(incomingHtlc: IncomingHtlc) extends CommitmentOutput
+  case class OutHtlc(outgoingHtlc: OutgoingHtlc) extends CommitmentOutput
 }
-case class DirectedHtlc(direction: Direction, add: UpdateAddHtlc) extends CommitmentOutput
+
+sealed trait DirectedHtlc {
+  def direction: Direction
```

Comment:
Do we really need to keep `Direction`? It seems like we can go further and remove that entirely, and rely on pattern matching only.